systemtest/src/test/java/io/strimzi/systemtest/matchers/IsLogHasUnexpectedErrors.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.matchers;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+
+/**
+ * <p>A IsLogHasUnexpectedErrors is custom matcher to check log form kubernetes client
+ * doesn't have any unexpected errors. </p>
+ */
+public class IsLogHasUnexpectedErrors extends BaseMatcher<String> {
+
+    @Override
+    public boolean matches(Object actualValue) {
+        if (actualValue != null && actualValue instanceof String && !actualValue.equals("")) {
+            for (LogWhiteList value : LogWhiteList.values()) {
+                if (((String) actualValue).contains(value.name)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void describeTo(Description description) {
+        description.appendText("The log should not contain unexpected errors.");
+    }
+
+    enum LogWhiteList {
+        CO_TIMEOUT_EXCEPTION("io.strimzi.operator.cluster.operator.resource.TimeoutException"),
+        VERTEX_EXCEPTION("io.vertx.core.VertxException: Thread blocked"),
+        NO_ERROR("NoError");
```

Comment:
@scholzj this is what I mentioned before. We have simple DEBUG message in CO log that looks like `2018-06-26 10:50:07 DEBUG DnsNameResolver:86 - [id: 0xb4cb3dde] RECEIVED: [6437: /10.96.0.10:53], DatagramDnsResponse(from: /10.96.0.10:53, to: /0.0.0.0:38890, 6437, QUERY(0), NoError(0), RD RA)\n2018-06-26 ` and in previous implementation test will be marked as failed because code executes `grep` command for CO log by keywords `Exception`, `Error`, `Throwable` and gets `NoError(0)` which is not really an error. So this is a reason that we have whitelist for errors and enums like `NO_ERROR`

Reply:
1. That _at least_ deserves a comment on `NO_ERROR` to explain what it's supposed to be whitelisting and why that's OK.

2. While a `VertxException: Thread blocked` doesn't _necessarily_ mean that the CO is in a bad state, the fact is that we simply shouldn't be blocking the event loop thread. So if we add this to the whitelist we will miss opportunities to fix places where we're accidentally blocking the event loop. So, in short, I don't think that should be whitelisted.

Reply:
@tombentley thanks for the review. TBH I didn't see #383 issue before and I thought that if `VertxException` is not related to Strimzi we can't fix it, but now I think otherwise, thanks